dev/org.eclipse.codewind.openapi.ui.test/src/org/eclipse/codewind/openapi/ui/test/BaseTestCase.java

```diff
@@ -23,30 +23,96 @@
 import org.eclipse.codewind.openapi.ui.test.menus.utils.UITestUtilities;
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
 import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.jobs.IJobManager;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jface.dialogs.IDialogConstants;
 import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
 import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.SWTBot;
 import org.eclipse.swtbot.swt.finder.SWTBotTestCase;
 import org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotButton;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotCombo;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotMenu;
 import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
 
 public class BaseTestCase extends SWTBotTestCase {
 
 	protected SWTWorkbenchBot swtWorkbenchBot = new SWTWorkbenchBot();
-	
+
 	protected IProject testProject;
 	protected IFile definitionFile;
-	
+
 	// Configurable options to test and should be overridden
 	protected String newProjectName = "TestProject";
 	protected String sourceDefinition = Constants.PETSTORE_30;
 	protected String targetDefinitionInProject = "petstore.yaml";
-	
+
 	public BaseTestCase() {
 		// Empty
 	}
-	
+
+	protected void testOpenAPIWizardHappyPath(String type, String language, String fileToCheck) {
```

Comment:
This method accepts `language` as a parameter, but the HTML generator has no language option.  Only the client and server generators/wizards have it.  Would it be better to have two methods?  One for the two language-based generators and one for HTML only?   The reason is that we might add more fields to the code gen wizard pages.  and those options will not apply to HTML.   Maybe the earlier we separate or make them distinct, the better?

Reply:
I am thinking we should keep these in one method.  Otherwise we are duplicating lots of things.  If the wizard has some changes in common, we would have to make multiple changes.

Reply:
Travis, we agreed this was ok.  We'll cross that bridge when we get to it.